controller/workitem.go

```diff
@@ -255,20 +250,32 @@ func (c *WorkitemController) Delete(ctx *app.DeleteWorkitemContext) error {
 	if err != nil {
 		return jsonapi.JSONErrorResponse(ctx, err)
 	}
-	authorized, err := authz.Authorize(ctx, wi.SpaceID.String())
+
+	// This checks if user is space admin or workitem creator or not
+	creator := wi.Fields[workitem.SystemCreator]
+	if creator == nil {
+		return jsonapi.JSONErrorResponse(ctx, errors.NewInternalError(ctx, errs.New("work item doesn't have creator")))
+	}
+	authorized, err := c.isWorkitemCreatorOrSpaceOwner(ctx, wi.SpaceID, creator.(string), currentUserIdentityID.String())
 	if err != nil {
-		return jsonapi.JSONErrorResponse(ctx, errors.NewUnauthorizedError(err.Error()))
+		forbidden, err := errors.IsForbiddenError(err)
```

Comment:
@DhritiShikhar I think the original `err` (the one from `isWorkitemCreatorOrSpaceOwner`) would be lost if you do a 
```go
forbidden, err := errors.IsForbiddenError(err)
```
and at line 266, we're returning `err` (which came from `IsForbiddenError`). So effectively we're returning the error from `IsForbiddenError` instead of `isWorkitemCreatorOrSpaceOwner`.

Can we do something like this?
```go
if err != nil {
    // We don't really need to check the error returned by isForbiddenError(). 
    // We'll just return the original error if this fails.
    if yes, _ := errors.IsForbiddenError(err); yes {
        return jsonapi.JSONErrorResponse(ctx, errors.NewForbiddenError("user is not authorized to delete the workitem")
    }
    return jsonapi.JSONErrorResponse(ctx, errors.NewInternalError(..., "failed to delete work item")
]
```